zerver/tests/test_messages.py

```diff
@@ -277,6 +277,47 @@ def test_bad_stream_id(self) -> None:
         result = self.client_get(endpoint, dict())
         self.assert_json_error(result, 'Invalid stream id')
 
+class TopicDeleteTest(ZulipTestCase):
+    def test_topic_delete(self) -> None:
+        user_profile = self.example_user('iago')  # Admin user
+        stream_name = 'new_stream'
+        topic_name = 'new topic 1'
+
+        # Create new stream
+        self.subscribe(user_profile, stream_name)
+
+        # Send message to new stream
+        stream = get_stream(stream_name, user_profile.realm)
+        self.send_stream_message(user_profile.email, stream_name, topic_name=topic_name)
+
+        # Deleting the new topic
+        self.login(user_profile.email, realm=user_profile.realm)
+        endpoint = '/json/streams/' + str(stream.id) + '/delete_topic'
+        result = self.client_post(endpoint, {
+            "topic_name": topic_name
+        })
+        self.assert_json_success(result)
```

Comment:
At the end, we should use e.g. `self.get_last_message` to check that the messages were actually deleted.

(And for the non-admin case, we should do the same to confirm they weren't).

This sort of assertion is really important for making a test validate that the right thing happened.